tcms/testruns/static/testruns/js/get.js

```diff
@@ -918,17 +918,27 @@ export function bindDeleteLinkButton () {
 function removeCases (testRunId, testCaseIds) {
     for (const testCaseId of testCaseIds) {
         jsonRPC('TestRun.remove_case', [testRunId, testCaseId], () => {
-            const tePK = $(`.test-execution-case-${testCaseId}`)
-                .find('.test-execution-checkbox')
-                .data('test-execution-id')
+            const testExecutionsIds = []
+            $(`.test-execution-case-${testCaseId}`)
+                .each(
+                    function () {
+                        testExecutionsIds.push(
+                            $(this)
+                                .find('.test-execution-checkbox')
+                                .data('test-execution-id')
+                        )
+                    }
+                )
```

Comment:
This is definitely wrong and has been wrong from the very beginning.  When you select test executions and delete them the expected behavior is that only items which are selected will be deleted, not everything else related to the same test case. 

IMO fixing the problem will require a few more steps, probably best to be done separately.

1) Copy `TestRun.remove_case()` API method under a new name `TestRun.remove_execution` and modify it to accept a dictionary (see methods which accept `values` parameters). That will needs some tests.
2) Adjust the UI and replace `TestRun.remove_case` with `TestRun.remove_execution` in all places where it is used, leaving the `.remove_case()` method to be used by external API consumers until it is removed. 

3) Come back to this PR and adjust as needed in order to allow adding parametrized test case variants for executions which have been previously removed! (if that is desired).

Reply:
I'm actually OK with this behavior, but I understand what you mean :)

At the moment I don't have time for that.